web3.js/src/connection.ts

```diff
@@ -3154,6 +3176,143 @@ export class Connection {
     return result;
   }
 
+  private async confirmTransactionUsingDurableNonceStrategy({
```

Comment:
This method is hundreds of lines long, can it be broken up a bit?

Reply:
This is how I understand when it's time to break up a function:

* It contains a testable subunit
* It contains something reusable

This function is only 100 lines (and really, it's just Prettier making it look even that long). The whole thing deals with durable nonces and none of that logic is reusable anywhere else. Theoretically, there are subunits in here that could be independently tested (eg. the expiry promise) but since it's all balled up in a JavaScript class there's no practical way to test it anyway.

On the _plus_ side, keeping all of this code together:

* means you can understand what it does from top to bottom, without having to jump out to other single-use functions
* means that if we ever extracted it into a separate module, there's only one thing to move.

¯\_(°ペ)_/¯

Reply:
I'm not sure what I was thinking here because this method isn't that long as you said. On another look, I agree that it's pretty self contained and doesn't need to broken up